dashboard/app/controllers/api_controller.rb

```diff
@@ -165,7 +165,7 @@ def user_progress_for_stage
 
     script = Script.get_from_cache(params[:script_name])
     stage = script.stages[params[:stage_position].to_i - 1]
-    script_level = stage.script_levels[params[:level_position].to_i - 1]
+    script_level = stage.cached_script_levels[params[:level_position].to_i - 1]
```

Comment:
As written, this now iterates over all script_levels in the script, rather than just those in the stage? Can this be avoided?